### PR TITLE
#67: Remove InstanceParameters from findInstallations, add flags to installable versions

### LINKED ISSUE #67: Add endpoint for getting parameters
When users list the installed extensions they don't need to know the parameters. In order to keep the response small, we should
* remove the parameters them from the list
* and provide a separate endpoint `GET /api/v1/extensionmanager/installations/{id}/{version}` that returns the parameters.

This should be implemented before https://github.com/exasol/extension-manager/issues/65

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #67

### CODE REVIEW

File: extensionAPI/extensionApi.go [hunk 1]
```diff
@@ -74,16 +74,23 @@ type installedExtension struct {
...
+	AddInstance             func(context *ExtensionContext, version string, params *ParameterValues) *JsExtInstance `json:"addInstance"`
+	FindInstances           func(context *ExtensionContext, version string) []*JsExtInstance                        `json:"findInstances"`
+	DeleteInstance          func(context *ExtensionContext, version, instanceId string)                             `json:"deleteInstance"`
+}
+
+type rawJsExtensionVersion struct {
+	Name       string `json:"name"`
+	Latest     bool   `json:"latest"`
+	Deprecated bool   `json:"deprecated"`
 }
```
[the remark this answers is not in the document]
Comment (kaklakariada) [reply extensionAPI/extensionApi.go:94]: Correct. This is the structure used by the JS extensions which could differ from the other structs.

File: extensionController/mocks.go [hunk 2]
```diff
@@ -81,6 +82,14 @@ func (mock *mockControllerImpl) GetAllExtensions(bfsFiles []BfsFile) ([]*Extensi
 	return nil, args.Error(1)
 }
 
+func (mock *mockControllerImpl) GetParameterDefinitions(tx *sql.Tx, extensionId string, extensionVersion string) ([]parameterValidator.ParameterDefinition, error) {
+	args := mock.Called(extensionId, extensionVersion)
+	if result, ok := args.Get(0).([]parameterValidator.ParameterDefinition); ok {
+		return result, args.Error(1)
```
[the remark this answers is not in the document]
Comment (kaklakariada) [reply extensionController/mocks.go:88]: This is the definition of the mock. In theory we could configure the mock to return both (result and error).

File: parameterValidator/parameterValidator.go [hunk 3]
```diff
@@ -15,6 +15,58 @@ import (
 //go:embed parameterValidator.js
 var dependencyValidatorJs string
 
+type ParameterDefinition struct {
+	Id            string
+	Name          string
+	RawDefinition map[string]interface{}
+}
+
+func ConvertDefinitions(rawDefinitions []interface{}) ([]ParameterDefinition, error) {
```
[the remark this answers is not in the document]
Comment (kaklakariada) [reply parameterValidator/parameterValidator.go:24]: The actual data structure is defined in TypeScript: https://github.com/exasol/extension-manager-interface/blob/main/src/parameters.ts#L5
For Extension Manager and clients this should be an opaque object that they can pass 1:1 to the validator (https://github.com/exasol/extension-parameter-validator/).
Additionally the type system of Go does not allow duplicating the types from TypeScript.

File: parameterValidator/parameterValidator.go [hunk 4]
```diff
@@ -15,6 +15,58 @@ import (
...
+	id, err = extractStringValue(def, "id")
+	if err != nil {
+		return
+	}
+	name, err = extractStringValue(def, "name")
+	if err != nil {
+		return
+	}
+	return
+}
```
[the remark this answers is not in the document]
Comment (kaklakariada) [reply parameterValidator/parameterValidator.go:58]: I changed the return statements to explicitly return values :)

File: parameterValidator/parameterValidator.go [hunk 5]
```diff
@@ -15,6 +15,58 @@ import (
...
+
+func extractStringValue(def map[string]interface{}, key string) (string, error) {
```
[the remark this answers is not in the document]
Comment (kaklakariada) [reply parameterValidator/parameterValidator.go:60]: No, there can also be bools and other values, see https://github.com/exasol/extension-manager-interface/blob/main/src/parameters.ts#L10

**Review by Nicoretti — APPROVED** [review]

**Review by Nicoretti — APPROVED** [review]

**Pull request merged**